### PR TITLE
APIM-98 add method to easily interrupt body or message flow

### PR DESCRIPTION
**Pull request by guillaumelamirand**
## Issue

https://graviteecommunity.atlassian.net/browse/APIM-98

## Description

Add method to easily interrupt body or message flow

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/apim-98-better-message-error-handling/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-xozdcolptw.chromatic.com)

### CODE REVIEW

File: gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/handler/context/AbstractExecutionContext.java [hunk 1]
```diff
@@ -60,7 +66,7 @@ public RS response() {
 
     @Override
     public Completable interrupt() {
-        return Completable.error(new InterruptionException());
+        return messagesInterruption();
```
Comment (jhaeyaert) [inline gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/handler/context/AbstractExecutionContext.java:69]: `messageInterruption` without the '**s**' ?

**Review by jhaeyaert — APPROVED** [review]
LGTM just one naming suggestion

File: gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/handler/context/AbstractExecutionContext.java (hunk 1 above)
Comment (guillaumelamirand) [reply gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/handler/context/AbstractExecutionContext.java:69]: this is used either by  interruptMessageS or interruptMessage. i would say both could work.

File: gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/handler/context/AbstractExecutionContext.java [hunk 2]
```diff
@@ -73,14 +79,47 @@ public Completable interruptWith(ExecutionFailure executionFailure) {
...
     @Override
     public Flowable<Message> interruptMessages() {
-        return Flowable.error(new InterruptionException());
+        return messagesInterruption().toFlowable();
+    }
+
+    @Override
+    public Maybe<Message> interruptMessage() {
+        return messagesInterruption().toMaybe();
+    }
```
Comment (ytvnr) [inline gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/jupiter/reactor/handler/context/AbstractExecutionContext.java:100]: same kind of comment as @jhaeyaert 
we have `interruptMessage` (singular) and `messagesInterruption()` (plural)

we also noticed the same "problem" in connector configuration and sometimes it's a little mindblowing or error prone.

maybe we should discuss it altogether to have a better understanding of when to put an "s" or not

**Review by ytvnr — APPROVED** [review]

**Pull request merged**